extensions/sequelize/src/sequelize/sequelize.datasource.base.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright LoopBack contributors 2022. All Rights Reserved.
+// Node module: @loopback/sequelize
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {LifeCycleObserver} from '@loopback/core';
+import {AnyObject, juggler} from '@loopback/repository';
+import debugFactory from 'debug';
+import {Dialect, Options as SequelizeOptions, Sequelize} from 'sequelize';
+import {
+  SupportedConnectorMapping as supportedConnectorMapping,
+  SupportedLoopbackConnectors,
+} from './connector-mapping';
+
+const debug = debugFactory('loopback:sequelize:datasource');
+const queryLogging = debugFactory('loopback:sequelize:queries');
+
+export class SequelizeDataSource
+  extends juggler.DataSource
+  implements LifeCycleObserver
+{
+  name: string;
+  constructor(public config: SequelizeDataSourceConfig) {
+    super(config);
```

Comment:
The issue seems to be caused by the Sequelize DataSource config being pass-through'd to Juggler DataSource.

Internally, Juggler DataSource will attempt to resolve the Connector itself, `sqlite`, among other things. `sqlite` is also one of the few connectors that Juggler DataSource will attempt to resolve with `loopback-connector-` prefix.

I'm curious if we can drop the dependence on Juggler DataSource, and instead implement greenfield only what's necessary. This is because quite a bit of Juggler is not actually used in LoopBack 4.

Reply:
Rifa, The dependency on Juggler DataSource is intentional as of now. 
But the point to note here is even if it's calling juggler data source the `sqlite3` package should be resolvable since it's there in package.json.

For example check the typeorm extension log here: https://github.com/loopbackio/loopback-next/pull/5110#issuecomment-755198271, There has no internal connection with juggler here but still it can't get the sqlite3 in CI pipeline.